jwst/resample/resample_spec.py

```diff
@@ -90,105 +91,298 @@ def build_interpolated_output_wcs(self, refmodel=None):
+            # mapping.inverse uses pix_to_dec.inverse.  The auto definition
+            # of mapping.inverse is to use the 2nd spatial coordinate, i.e. Dec.
+            if np.isclose(pix_to_dec.slope, 0, atol=1e-8):
+                mapping_tuple = (0, 1)
+                # Account for vertical or horizontal dispersion on detector
+                if spatial_axis:
+                    mapping.inverse = Mapping(mapping_tuple[::-1])
+                else:
+                    mapping.inverse = Mapping(mapping_tuple)
+
+            # The final transform
+            transform = mapping | (pix_to_ra & pix_to_dec | undist2sky) & pix_to_wavelength
+
+            det = cf.Frame2D(name='detector', axes_order=(0, 1))
+            sky = cf.CelestialFrame(name='sky', axes_order=(0, 1),
+                                    reference_frame=coord.ICRS())
+            spec = cf.SpectralFrame(name='spectral', axes_order=(2,),
+                                    unit=(u.micron,), axes_names=('wavelength',))
+            world = cf.CompositeFrame([sky, spec], name='world')
+
+            pipeline = [(det, transform),
+                        (world, None)]
+
+            output_wcs = WCS(pipeline)
+
+            # compute the output array size in WCS axes order, i.e. (x, y)
+            output_array_size = [0, 0]
+            output_array_size[spectral_axis] = len(wavelength_array)
+            output_array_size[spatial_axis] = len(x_tan_array)
+
+            # turn the size into a numpy shape in (y, x) order
+            self.data_size = tuple(output_array_size[::-1])
+            bounding_box = resample_utils.wcs_bbox_from_shape(self.data_size)
+            output_wcs.bounding_box = bounding_box
+
+          
+        else: 
+        # for each input model convert slit x,y to ra,dec,lam
+        # for the center of the slit compute ra,dec held in
+        # x_tan, y_tan
+        # append the x_tan, y_tan and wavelength arrays
+        # sort them and store in lookup table 
+            print('using new method to determine output WCS')
+            all_wavelength = []
+            all_ra_slit = []
+            all_dec_slit = []
+            xtan_slope = []
+            ytan_slope = []
+            for im, model in enumerate(self.input_models):
+                wcs = model.meta.wcs
+                bb = wcs.bounding_box
+                grid = wcstools.grid_from_bounding_box(bb)
+                ra, dec, lam = np.array(wcs(*grid))
+
+                spectral_axis = find_dispersion_axis(model)
+                spatial_axis = spectral_axis ^ 1
+            # print('spectral axis',spectral_axis)
+            # print('spatial axis',spatial_axis)
+            # Compute the wavelength array, trimming NaNs from the ends
+                wavelength_array = np.nanmedian(lam, axis=spectral_axis)
+                wavelength_array = wavelength_array[~np.isnan(wavelength_array)]
+                all_wavelength.append(wavelength_array)
```

Comment:
I don't think we can just append here.  It is important to not just have the wavelengths, but to know the points (pixel positions) that map to those wavelengths)